### ISSUE TITLE
Some indices are not returned in `knn_interpolate`

### ISSUE THREAD

**Issue by saedrna** [pyg-team/pytorch_geometric#6084]
### 🐛 Describe the bug

**First of all, I have to say I tried to make a toy example to reproduce the bug, but failed. So sorry for the inconvenience.**

https://github.com/pyg-team/pytorch_geometric/blob/6aa454a7de1e4d1ed733d7f6ca65f1cf50ca6226/torch_geometric/nn/unpool/knn_interpolate.py#L46-L57

When my point clouds have some outlier with large position coordinates, they are not returned in the `y_idx, x_idx`. Here is the debug results, you can see that `pos_y` has dim of 80000, but only 79997 indices are returned. 

![Screenshot_2022-11-28_20-48-45](https://user-images.githubusercontent.com/3850191/204285607-044e9321-b63c-4dee-8829-db260c8b2eb3.jpg)

This cause some features in `y` are `nan` values.

![Screenshot_2022-11-28_20-56-08](https://user-images.githubusercontent.com/3850191/204285751-c8664a3c-221c-42aa-85ed-c6f20f987998.jpg)

The coordinates in `pos_y` is below.

```
tensor([[ 178.2011, -146.3600, 1391.0408],
        [ -18.2090,  -54.5212,  995.9201],
        [ -18.6603,  -54.4893,  996.0385]], device='cuda:0')
```

I am not sure if this is a bug or intended behavior.

Han

### Environment

* PyG version: 2.1.0
* PyTorch version: 1.11
* OS: Linux Mint
* Python version: 3.8
* CUDA/cuDNN version: 11.3.0
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda
* Any other relevant information (*e.g.*, version of `torch-scatter`): pytorch-scatter 2.0

**Comment by rusty1s** [comment]
Thanks for reporting. Is it possible to normalize the `pos` (e.g., to interval `[-1, 1]`) and see if the same issue still exists? There might be some weird overflows happening.

**Comment by saedrna** [comment]
I used a traditional technique to remove the outlier points during preprocessing step and the model can now train without problems.
Normalization may not be a good idea for LiDAR point clouds (although I think this will also by-pass this issue), because the actual metric sizes of the objects are very important.

**Comment by rusty1s** [comment]
Ok, good that you are unblocked. If you by chance have any reproducible example, I am happy to look closer into this :)
